standards/core/lightpush.md

```diff
@@ -0,0 +1,88 @@
+---
+slug: 19
+title: 19/WAKU2-LIGHTPUSH
+name: Waku v2 Light Push
+status: draft
+editor: Zoltan Nagy <zoltan@status.im> 
+contributors: 
+  - Hanno Cornelius <hanno@status.im>
+  - Daniel Kaiser <danielkaiser@status.im>
+  - Oskar Thorén <oskarth@titanproxy.com>
+---
+
+previous version: `/vac/waku/lightpush/2.0.0-beta1` [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/waku/standards/core/19/lightpush)
+
+---
+**Protocol identifier**: `/vac/waku/lightpush/2.0.0-beta2`
+
+## Motivation and Goals
+
+Light nodes with short connection windows and limited bandwidth wish to publish messages into the Waku network.
+Additionally, there is sometimes a need for confirmation that a message has been received "by the network"
+(here, at least one node).
+
+`19/WAKU2-LIGHTPUSH` is a request/response protocol for this.
+
+## Payloads
+
+```protobuf
+syntax = "proto3";
+
+message LightPushRequest {
+    enum RequestType { // Future extension point
+        RELAY = 0;
+    }
+    
+    string request_id = 1;
+    RequestType kind = 10;  // default is RELAY, placeholder for future extensions
+    string pubsub_topic = 20;
+    WakuMessage message = 21;
+}
+
+message LightPushResponse {
+    enum Status {
+        SUCCESS = 0;
+        BAD_REQUEST = 400;
+        NO_PEERS_TO_RELAY = 404;
+        PAYLOAD_TOO_LARGE = 413;
+        UNSUPPORTED_TOPIC = 415;
+        TOO_MANY_REQUESTS = 429;
+        INTERNAL_SERVER_ERROR = 500;
+        SERVICE_UNAVAILABLE = 503;    
+    }
```

Comment:
What if instead of defining the codes we keep this impl. detail? We can still specify that error codes should follow http standards.

Reply:
Hm, interesting idea not to have them in the communication interface description what outcome you might prepare for. For me it is more should be here than not.

Reply:
Indeed, for new versions of protocols (and after a number of discussions), we've opted to leave `status_code` and `status_desc` generic `uint32` and `string` fields in order to define the possible error codes in the implementation. The reason being that we extend this list quite often and we'll soon get in a situation where the implementation and spec diverge. Of course, once we believe we have some stability in the services, we may publish a stand-beside spec with the list of supported error codes. See https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/12/filter.md#protobuf for how we're doing it in the new filter (and new store) protocols.

Reply:
Done. Added into a separate section as examples.